src/main/java/bio/terra/common/logging/GoogleJsonLayout.java

```diff
@@ -133,11 +141,21 @@ public void stop() {
       for (Object arg : event.getArgumentArray()) {
         try {
           if (arg instanceof Map) {
+            // Handle arbitrary Map by splatting each key-value pair into the main output map.
             Map<String, Object> jsonMap = (Map<String, Object>) arg;
             jsonMap.forEach(map::put);
           } else if (arg instanceof JsonNode) {
+            // Handle Jackson JsonNode by splatting each property sub-tree into the main output map.
             JsonNode jsonNode = (JsonNode) arg;
             jsonNode.fields().forEachRemaining(entry -> map.put(entry.getKey(), entry.getValue()));
+          } else if (arg instanceof JsonObject) {
+            // Some libraries use GSON rather than Jackson for arbitrary JSON data, and we should
+            // support that too. Since we're using Jackson for top-level serialization, we need to
+            // convert this gson.JsonObject into a jackson.databind.JsonNode for storage in the
+            // output map. The simplest way to do this is by using a JSON string as intermediary.
+            JsonObject jsonObject = (JsonObject) arg;
+            JsonNode jsonNode = objectMapper.readTree(jsonObject.toString());
+            jsonNode.fields().forEachRemaining(entry -> map.put(entry.getKey(), entry.getValue()));
           }
```

Comment:
Log/warn if an unexpected type shows up here. Indicate if there are other types that intentionally aren't handled.

Reply:
For my own confusion, maybe just an example of what happens to a JsonNode. Does it become a `Map<String, JsonNode>`? If so, how is that any better?

And you're not trying to convert all POJO types into JSON, which I originally thought was a goal for some reason.

Reply:
The best way I found to describe it is in the comment above the prior if-else: "splat each property sub-tree into the main output map".